lmdeploy/pytorch/weight_loader/dist_utils.py

```diff
@@ -0,0 +1,275 @@
+                                   rank: int,
+                                   world_size: int,
+                                   prefix: str = ''):
+    """colwise parallelize loralinear."""
+    rowwise_parallelize_linear_naive(module.base_layer,
+                                     loader,
+                                     rank=rank,
+                                     world_size=world_size,
+                                     prefix=prefix)
+    for key, mod in module.lora_A.items():
+        ada_loader = loader.adapter(key)
+        rowwise_parallelize_linear_naive(mod,
+                                         ada_loader,
+                                         rank=rank,
+                                         world_size=world_size,
+                                         prefix=get_prefixed_name(
+                                             'lora_A', prefix))
+    for key, mod in module.lora_B.items():
+        ada_loader = loader.adapter(key)
+        colwise_parallelize_linear_naive(mod,
+                                         ada_loader,
+                                         rank=rank,
+                                         world_size=world_size,
+                                         prefix=get_prefixed_name(
+                                             'lora_B', prefix))
+    module._tp_mode = 'colwise'
+
+
+def rowwise_parallelize_linear(module: torch.nn.Module,
+                               loader: ModelWeightLoader,
+                               rank: int,
+                               world_size: int,
+                               prefix: str = ''):
+    """colwise parallelize linear."""
+    if isinstance(module, (torch.nn.Linear, QLinear)):
+        return rowwise_parallelize_linear_naive(module,
+                                                loader,
+                                                rank=rank,
+                                                world_size=world_size,
+                                                prefix=prefix)
+    elif isinstance(module, LoRALinear):
+        return rowwise_parallelize_loralinear(module,
+                                              loader,
+                                              rank=rank,
+                                              world_size=world_size,
+                                              prefix=prefix)
+    else:
+        raise TypeError(f'Unsupported module: {type(module)}')
+
+
+def colwise_split_parallelize_linear_naive(module: torch.nn.Module,
+                                           sections: List[int],
+                                           loader: ModelWeightLoader,
+                                           rank: int,
+                                           world_size: int,
+                                           prefix: str = ''):
+    """colwise split linear naive."""
+    for name, param in module.named_parameters():
+        dtype = param.dtype
```

Comment:
It seems the for-named-parameters block is similar to the for-named-buffers block.
Can they be merged together?